components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2113,8 +2113,12 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    protected defaultSpendingLimit = 100;
-    async subscribeToStripe(ctx: TraceContext, attributionId: string, setupIntentId: string): Promise<void> {
+    async subscribeToStripe(
+        ctx: TraceContext,
+        attributionId: string,
+        setupIntentId: string,
+        usageLimit: number,
+    ): Promise<number | undefined> {
```

Comment:
FWIW, this API is now super awkward in terms of the response. A consumer wouldn't expect to get a number as a response to `subscribeToStripe`. `subscribe` should return a Subscription, which can have a property of `usageLimit`.